zebra-scan/src/service/scan_task/scan.rs

```diff
@@ -326,19 +355,23 @@ pub async fn scan_height_and_store_results(
         let sapling_tree_size = 1 << 16;
 
         tokio::task::spawn_blocking(move || {
-            let dfvk_res =
-                scan_block(&network, &block, sapling_tree_size, &dfvks).map_err(|e| eyre!(e))?;
-            let ivk_res =
-                scan_block(&network, &block, sapling_tree_size, &ivks).map_err(|e| eyre!(e))?;
+            // Convert the keys to zcash_client_backend new format for `scan_block` function.
+            // TODO: This is probabably wrong, fix it.
+            let ready_scan_block_keys = ready_scan_block_keys(parsed_keys);
+            let ready_scan_block_keys = ready_scan_block_keys
+                .iter()
+                .next()
+                .expect("we should have 1 key")
+                .1;
```

Comment:
I think just scanning with the first key is ok here but i need to confirm.

Reply:
Yes, that's correct. The fn `scan_block` currently works only with ivks anyway.